File: client/src/app/model/department/Department.ts
import {LecturerItem} from 'src/app/model/lecturer/LecturerItem';

export class Department {
  id = 0;
  name = '';
  facultyId = 0;
  email = '';
  phone = '';
  info = '';
  lecturers: LecturerItem[] = [];
}

File: graphql-service/src/main/java/org/lnu/timetable/model/department/Department.java
package org.lnu.timetable.model.department;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.util.List;

@Data
@Table("departments")
public class Department {

    public static final List<String> selectableFields = List.of(
        "name",
        "facultyId",
        "email",
        "phone",
        "info"
    );

    @Id
    private Long id;

    private String name;

    private Long facultyId;

    private String email;

    private String phone;

    private String info;

}
